script.js

```diff
@@ -1,12 +1,209 @@
 //You can edit ALL of the code here
+let input = document.getElementById("search");
+let container = document.getElementById("container");
+let select = document.getElementById("select-input")
+let option = document.getElementById("option");
+let episodesOption = document.getElementById("episodes-option")
+let rootDiv = document.getElementById("root");
+let countDisplay = document.getElementById("count-display");
+let showSelect = document.getElementById("shows");
+let showsNav = document.getElementById("shows-nav");
+let searchShows = document.getElementById("search-shows");
+let showOption;
+let showEpisodes = [];
+
+
+function setup() {
+  
+  populateShowsDropDown();
+  select.style.display = "none";
+}
+
+ // function to load all episodes
+function makePageForEpisodes(episodeList) {
+  container.innerHTML = "";
+  countDisplay.innerHTML = `Displaying ${episodeList.length} of ${showEpisodes.length}`
+  episodeList.forEach((episode) => {
+    container.innerHTML += `<div id="episodesDiv"> <div id="title-div"> <h2>${episode.name} 
+    - Season: ${paddedToTwoDigits(episode.season)} 
+    - Episode: ${paddedToTwoDigits(episode.number)} </h2> </div> 
+    <img id="img" src="${episode.image != null ? episode.image.medium : ''}"/> 
+    ${episode.summary}<div id="level-500"> - Runtime: ${episode.runtime}</div></div>`;
+});
+ input.style.display = "block"; 
+}
+
+// function to load shows
+function makePageForShows(showList) {
+  container.innerHTML = "";
+  countDisplay.innerHTML = "";
+  showList.forEach((episode) => {
+    container.innerHTML += `<div onclick="fetchEpisodes(${episode.id})" id="episodesDiv"> <div id="title-div"> <h2>${episode.name} </h2> </div> 
+    <img id="img" src="${episode.image != null ? episode.image.medium : ''}"/> 
+    ${episode.summary}<div id="level-500">Genres: ${episode.genres} - Status: ${episode.status} - Rating: ${episode.rating.average} - Runtime: ${episode.runtime}</div></div>`;
+});
+  
+}
+ 
+// function zero-padded to two digits.
+function paddedToTwoDigits(num) {
+  return num.toString().padStart(2, 0);
+}
+
+// function to search for episodes
+function search(elem) {
+  let searchString = elem.target.value.toLowerCase();
+  let filteredEpisodes = showEpisodes.filter((episode) => {
+    return (
+      episode.name.toLowerCase().includes(searchString) ||
+      (episode.summary && episode.summary.toLowerCase().includes(searchString))
+    );
+  });
+  container.innerHTML = "";
+  makePageForEpisodes(filteredEpisodes);
+}
+input.style.display = "none";
+input.addEventListener("input", search);
```

Comment:
I'd avoid having global scope code like this as it is hard to read and makes debugging harder. I would move this into the setup function.